### PR TITLE
Re-enabling redefined-builtin for Pylint.

### PR DESCRIPTION
**Pull request by dhermes**
In the process, refactoring places where builtins were used intentionally or un-intentionally.

Inspired by discussion with @rimey in #1691.

### CODE REVIEW

File: gcloud/exceptions.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ class InternalServerError(ServerError):
     code = 500
 
 
-class NotImplemented(ServerError):
+class NotImplementedMethod(ServerError):
     """Exception mapping a '501 Not Implemented' response."""
```
Comment (tseaver) [inline gcloud/exceptions.py]: ISTM that `MethodNotImplemented` is a clearer mapping of the HTTP response code.

File: gcloud/search/document.py [hunk 2]
```diff
@@ -204,8 +204,9 @@ def _parse_value_resource(resource):
             return TimestampValue(value)
         if 'geoValue' in resource:
             lat_long = resource['geoValue']
-            lat, long = [float(coord.strip()) for coord in lat_long.split(',')]
-            return GeoValue((lat, long))
+            lat, long_val = [float(coord.strip())
+                             for coord in lat_long.split(',')]
+            return GeoValue((lat, long_val))
```
Comment (tseaver) [inline gcloud/search/document.py]: wouldn't `latitude` and `longitude` be clearer?  Or even `lat` and `lon`.

File: gcloud/exceptions.py (hunk 1 above)
Comment (dhermes) [inline gcloud/exceptions.py]: On it.

File: gcloud/search/document.py (hunk 2 above)
Comment (dhermes) [inline gcloud/search/document.py]: Will do.

**Comment by dhermes** [comment]
@tseaver PTAL

**Comment by tseaver** [comment]
LGTM

**Comment by dhermes** [comment]
Will squash then merge.

**Pull request merged**